### PR TITLE
[fix](nereids) fix bug in PullUpJoinFromUnion rule

### PR DESCRIPTION
**Pull request by englefly**
## Proposed changes
some slots are mapped to wrong place

Issue Number: close #xxx

### CODE REVIEW

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/PullUpJoinFromUnion.java [hunk 1]
```diff
@@ -403,8 +404,9 @@ private boolean tryMapJoinSlotToUnion(LogicalUnion union, List<Pair<LogicalJoin<
                     }
```
Comment (XieJiann) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/PullUpJoinFromUnion.java:404]: You should check whether the expr is in regular child output in loc 187

Comment (englefly) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/PullUpJoinFromUnion.java:404]: 如果expr 不在regular 里面，那就不应该有这个expr了吧

Comment (XieJiann) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/PullUpJoinFromUnion.java:404]: 有可能是被regular裁剪掉了

**Comment by englefly** [comment]
run buildall

**Pull request closed without merging**